### PR TITLE
feat: First start after update will start with a flag

### PR DESCRIPTION
**Pull request by maczikasz**
After updating with the auto updater, the first start will start with a "--updated" flag

### CODE REVIEW

**Comment by maczikasz** [comment]
I didn't add tests yet because I wasn't sure where it would make most sense, and how much should I modify in the original files or how much stuff can I add for this very small change

**Comment by develar** [comment]
Please modify https://github.com/electron-userland/electron-builder/blob/master/templates/nsis/boringInstaller.nsh#L10 also (sorry, I didn't mentioned this place).

**Comment by develar** [comment]
> I didn't add tests yet

Yes. it is a little bit complicated and not enabled by default. Please see https://github.com/electron-userland/electron-builder/blob/master/test/src/windows/nsisTest.ts#L86

Writing test will be much more complex than implementation.

Have you tested it manually?

**Comment by maczikasz** [comment]
I did test it manually and it worked

I will add to the boring installer, I thought it was on purpose, because if you are running an update you wouldn't want the boring updater, but yeah also possible.

One other thing I was thinking about, but that's probably not part of the PR just an idea and I am interested in your opinion, if you do an update, do you really want to show the licence again?

**Comment by develar** [comment]
> do you really want to show the licence again

Nice catch. Well... What if license was changed? But in general, 👍  In most cases it makes sense to not show license again. The same as in case of macOS — if dmg has license, it is not included into the zip package update.

**Comment by maczikasz** [comment]
I changed the boring installer (still only manual tests)

Would you want me to implement the lincence thing or would you'd like to think about a smarter solution (changing licence or smth)

**Comment by develar** [comment]
> Would you want me to implement the lincence thing

Yes, simple check is enough. http://nsis.sourceforge.net/Skipping_Pages

**Comment by maczikasz** [comment]
I added the skipping and tested manually (both boring and oneClick)

File: templates/nsis/boringInstaller.nsh [hunk 1]
```diff
@@ -7,13 +7,28 @@
...
+    ${GetParameters} $R0
+    ${GetOptions} $R0 "--update" $R1
+    ${IfNot} ${Errors}
+      ${StdUtils.ExecShellAsUser} $0 "$SMPROGRAMS\${PRODUCT_FILENAME}.lnk" "open" "--updated"
+    ${Else}
+      ${StdUtils.ExecShellAsUser} $0 "$SMPROGRAMS\${PRODUCT_FILENAME}.lnk" "open" ""
+    ${endif}
+  FunctionEnd
+
+  Function licensePre
```
Comment (develar) [inline templates/nsis/boringInstaller.nsh]: Could you please move `licensePre` function to [common.nsh](https://github.com/electron-userland/electron-builder/blob/master/templates/nsis/common.nsh) (after line `!ifndef BUILD_UNINSTALLER`)?

Comment (maczikasz) [inline templates/nsis/boringInstaller.nsh]: sure

Comment (maczikasz) [inline templates/nsis/boringInstaller.nsh]: Error: resolving install function "licensePre" in function "mui.LicensePre_22.5.19"
 it doesn't work any more, do I need to do something like export or change visiblity?

Comment (develar) [inline templates/nsis/boringInstaller.nsh]: Strange. Ok, leave it as is.

**Comment by develar** [comment]
> warning: install function "licensePre" not referenced - zeroing code (306-315) out
> https://circleci.com/gh/electron-userland/electron-builder/352?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

Please wrap function into `!ifdef LICENSE_FILE` to fix this warning.

**Comment by maczikasz** [comment]
Done I moved it to the already existing ifdef

**Comment by develar** [comment]
Thanks a lot! Nice feature.

**Comment by maczikasz** [comment]
Could you pls release a version of "electron-auto-updater" that contains change?

**Comment by develar** [comment]
electron-auto-updater@0.7.0 published.

**Comment by maczikasz** [comment]
awesom thanks vm

**Pull request merged**